src/services/completions.ts

```diff
@@ -2830,4 +2832,15 @@ namespace ts.Completions {
     function isStaticProperty(symbol: Symbol) {
         return !!(symbol.valueDeclaration && getEffectiveModifierFlags(symbol.valueDeclaration) & ModifierFlags.Static && isClassLike(symbol.valueDeclaration.parent));
     }
+
+    function tryGetObjectLiteralContextualType(node: ObjectLiteralExpression, typeChecker: TypeChecker) {
+        const type = typeChecker.getContextualType(node);
+        if (type) {
+            return type;
+        }
+        if (isBinaryExpression(node.parent) && node.parent.operatorToken.kind === SyntaxKind.EqualsToken) {
+            return typeChecker.getTypeAtLocation(node.parent);
+        }

```

Comment:
Technically we're relying on the fact that there will be a contextual type on the RHS of an `=`, so that we must be on the left side of the `=`. Not sure if that's an issue long-term.

Reply:
You are right. I can change to `typeChecker.getTypeAtLocation(node.parent.right);`. Do I need to do it?

Reply:
It's not so much that - the `=` will always return the type of the right operand. It's more that you want to check that `node.parent.left === node`, but it's up to you. I don't think it'll be a problem long-term.